### DISCUSSION TITLE
Guest Book built with JS SDK and ETH Shanghai workshop

### DISCUSSION THREAD

**Discussion by think-in-universe** [near/near-sdk-js#59] [Show and tell]
We have a lesson talking about NEAR JavaScript SDK this week in our [CN NEAR Certified Developer course in May](https://segmentfault.com/a/1190000041844009).  

- slides of the NCD lesson: [_Writing Smart Contracts in JavaScript_](https://github.com/near-x/ncd-cn/blob/master/cohorts/ncd-cn-8/slides/NEAR%20%E8%AE%A4%E8%AF%81%E5%BC%80%E5%8F%91%E8%80%85%EF%BC%8826%EF%BC%89%EF%BC%9A%20%E7%94%A8%20JavaScript%20%E5%BC%80%E5%8F%91%20NEAR%20%E6%99%BA%E8%83%BD%E5%90%88%E7%BA%A6.pdf)
- [simplified slides](https://bit.ly/near-js-contract) for ETH Shanghai and the [recording](https://youtu.be/Xmm5fRvbJ0U)

During the lesson, we implemented the Guest Book demo with JS SDK. 

- Here is the PR demos how it looks like: https://github.com/near-examples/guest-book/pull/702
- Preview version at: https://think-in-universe.github.io/guest-book/

Please feel free to take a look and share your comments. 

There are some caveats that are not easy to notice at the beginning, so it takes some time for us to debug and resolve the issues.

**Comment by volovyks** [comment]
@think-in-universe reviewed.
You can also use [this](https://github.com/near/near-sdk-js-template-project) template project, but it has no front end yet.
